cmake_modules/FindLIBMEMCACHED.cmake

```diff
@@ -21,47 +21,131 @@
+    endif()
+
+    # Search for native library to build against
+    IF (WIN32)
+        SET (libmemcached_lib "libmemcached")
+        SET (libmemcachedUtil_lib "libmemcachedutil")
     ELSE()
-      SET (LIBMEMCACHED_VERSION_OK 0)
-      SET(MSG "libmemcached version '${LIBMEMCACHED_VERSION_STRING}' incompatible with min version>=${LIBMEMCACHED_FIND_VERSION}")
+        SET (libmemcached_lib "memcached")
+        SET (libmemcachedUtil_lib "memcachedutil")
+    ENDIF()
+
+    FIND_PATH(LIBMEMCACHED_INCLUDE_DIR libmemcached/memcached.hpp PATHS /usr/include /usr/share/include /usr/local/include PATH_SUFFIXES libmemcached)
+
+    FIND_LIBRARY (LIBMEMCACHEDCORE_LIBRARY NAMES ${libmemcached_lib} PATHS /usr/lib usr/lib/libmemcached /usr/share /usr/lib64 /usr/local/lib /usr/local/lib64)
+    FIND_LIBRARY (LIBMEMCACHEDUTIL_LIBRARY NAMES ${libmemcachedUtil_lib} PATHS /usr/lib /usr/share /usr/lib64 /usr/local/lib /usr/local/lib64)
+
+    #IF (LIBMEMCACHED_LIBRARY STREQUAL "LIBMEMCACHED_LIBRARY-NOTFOUND")
+    #  SET (LIBMEMCACHEDCORE_LIBRARY "")    # Newer versions of libmemcached are header-only, with no associated library.
+    #ENDIF()
+
+    SET (LIBMEMCACHED_LIBRARIES ${LIBMEMCACHEDCORE_LIBRARY} ${LIBMEMCACHEDUTIL_LIBRARY})
+
+    IF(LIBMEMCACHED_INCLUDE_DIR)
+        FILE (STRINGS "${LIBMEMCACHED_INCLUDE_DIR}/libmemcached-1.0/configure.h" version REGEX "#define LIBMEMCACHED_VERSION_STRING")
+        STRING(REGEX REPLACE "#define LIBMEMCACHED_VERSION_STRING " "" version "${version}")
+        STRING(REGEX REPLACE "\"" "" version "${version}")
+        SET (LIBMEMCACHED_VERSION_STRING ${version})
+        IF ("${LIBMEMCACHED_VERSION_STRING}" VERSION_EQUAL "${LIBMEMCACHED_FIND_VERSION}" OR "${LIBMEMCACHED_VERSION_STRING}" VERSION_GREATER "${LIBMEMCACHED_FIND_VERSION}")
+            SET (LIBMEMCACHED_VERSION_OK 1)
+            SET (MSG "${DEFAULT_MSG}")
+        ELSE()
+            SET (LIBMEMCACHED_VERSION_OK 0)
+            SET(MSG "libmemcached version '${LIBMEMCACHED_VERSION_STRING}' incompatible with min version>=${LIBMEMCACHED_FIND_VERSION}")
+        ENDIF()
     ENDIF()
-  ENDIF()
-      
-  include(FindPackageHandleStandardArgs)
-  find_package_handle_standard_args(libmemcached ${MSG}
-    LIBMEMCACHED_LIBRARIES
-    LIBMEMCACHED_INCLUDE_DIR
-    LIBMEMCACHED_VERSION_OK
-  )
-
-  MARK_AS_ADVANCED(LIBMEMCACHED_INCLUDE_DIRS LIBMEMCACHED_LIBRARIES)
+
+    include(FindPackageHandleStandardArgs)
+    find_package_handle_standard_args(libmemcached ${MSG}
+        LIBMEMCACHED_LIBRARIES
+        LIBMEMCACHED_INCLUDE_DIR
+        LIBMEMCACHED_VERSION_OK
+        )
+
+    if(NOT LIBMEMCACHED_FOUND AND MEMCACHED_USE_EXTERNAL_LIBRARY)
+        # Currently libmemcached versions are not sufficient on ubuntu 12.04 and 14.04 LTS
+        # until then, we build the required libraries from source
+        set(LIBMEMCACHED_URL https://launchpad.net/libmemcached/1.0/${LIBMEMCACHED_VERSION}/+download/libmemcached-${LIBMEMCACHED_VERSION}.tar.gz)
```

Comment:
Instead of using LIBMEMCACHED_FIND_VERSION we are using LIBMEMCACHED_VERSION which is set by the user.  LIBMEMCACHED_FIND_VERSION is the minimum version, and we now check to make sure that the downloaded and built version meets the same version requirements.